x-pack/solutions/security/plugins/security_solution/server/lib/entity_analytics/privilege_monitoring/data_sources/sync/index_sync.ts

```diff
@@ -0,0 +1,179 @@
+
+  const bulkUtilsService = createBulkUtilsService(dataClient);
+  const findStaleUsers = findStaleUsersForIndexFactory(dataClient);
+
+  const searchService = createSearchService(dataClient);
+
+  /**
+   * Synchronizes users from monitoring index sources and soft-deletes (mark as not privileged) stale entries.
+   *
+   * This method:
+   * - Retrieves all saved objects of type 'index' that define monitoring sources.
+   * - For each valid source with an index pattern, fetches usernames from the monitoring index.
+   * - Identifies users no longer present in the source index (stale users).
+   * - Performs a bulk soft-delete (marks as not privileged) for all stale users found.
+   * - Handles missing indices gracefully by logging a warning and skipping them.
+   *
+   * Additionally, all users from index sources are synced with the internal privileged user index,
+   * ensuring each user is either created or updated with the latest data.
+   *
+   * @returns {Promise<void>} Resolves when synchronization and soft-deletion are complete.
+   */
+  const plainIndexSync = async (soClient: SavedObjectsClientContract) => {
+    const monitoringIndexSourceClient = new MonitoringEntitySourceDescriptorClient({
+      soClient,
+      namespace: deps.namespace,
+    });
+    // get all monitoring index source saved objects of type 'index'
+    const indexSources = await monitoringIndexSourceClient.findByIndex();
+    if (indexSources.length === 0) {
+      dataClient.log('debug', 'No monitoring index sources found. Skipping sync.');
+      return;
+    }
+    const allStaleUsers: PrivMonBulkUser[] = [];
+
+    for (const source of indexSources) {
+      // eslint-disable-next-line no-continue
+      if (!source.indexPattern) continue; // if no index pattern, skip this source
+      const index: string = source.indexPattern;
+
+      try {
+        const batchUserNames = await syncUsernamesFromIndex({
+          indexName: index,
+          kuery: source.filter?.kuery,
+        });
+
+        // collect stale users
+        const staleUsers = await findStaleUsers(index, batchUserNames);
+        allStaleUsers.push(...staleUsers);
+      } catch (error) {
+        if (
+          error?.meta?.body?.error?.type === 'index_not_found_exception' ||
+          error?.message?.includes('index_not_found_exception')
+        ) {
+          dataClient.log('warn', `Index "${index}" not found — skipping.`);
+          // eslint-disable-next-line no-continue
+          continue;
+        }
+        dataClient.log(
+          'error',
+          `Unexpected error during sync for index "${index}": ${error.message}`
+        );
+      }
+    }
+    // Soft delete stale users
+    dataClient.log('debug', `Found ${allStaleUsers.length} stale users across all index sources.`);
+    if (allStaleUsers.length > 0) {
+      const ops = bulkUtilsService.bulkSoftDeleteOperations(allStaleUsers, dataClient.index);
+      await esClient.bulk({ body: ops });
```

Comment:
Probably not related, more like an FYI: 
I added this to my batch to ensure the index is refreshed before the task completes.
```suggestion
      await esClient.bulk({ body: ops, refresh: true });
```